aries_cloudagent/anoncreds/default/did_web/registry.py

```diff
@@ -112,3 +112,7 @@ async def update_revocation_list(
     ) -> RevListResult:
         """Update a revocation list on the registry."""
         raise NotImplementedError()
+
+    async def txn_submit(self, profile: Profile, ledger_transaction: str) -> str:
+        """Submit a transaction to the ledger."""
+        raise NotImplementedError()
```

Comment:
This addition is a red flag; `did:web` doesn't have a ledger or transactions.

Reply:
Ok. Thanks for doing a review. This came from wanting the anoncreds registry wanting to do the transaction submission from Ian's comment. I was thinking it could just implement it to do nothing if that was the case for did:web. 

The most confusing thing for me still is what should happen in existing acapy and what should happen from the anoncreds registry.

Reply:
The presence (or absence) of a ledger in use by anoncreds should be hidden behind the anoncreds interface and be a detail of a specific registry implementation. I'm between meetings so I haven't had a chance to look closer yet at the goal of the method to make alternate recommendations but I should be able to look closer later!

Reply:
The alternative is to move the ledger write from within the transaction manager (here: https://github.com/hyperledger/aries-cloudagent-python/pull/2715/files#diff-7e8dcd5a6024ec4d34242c09d77451ed6c5309435710e4278eb82ebbfce65c2eR424) to the event listener in the anoncreds issuer (here: https://github.com/hyperledger/aries-cloudagent-python/pull/2715/files#diff-fad60a24ebdabddf55dab0b471caef11c0ab82243fef6e0f1df2f9f250e6acb6R680)

So ... within the transaction manager, it just needs to know enough to *not* try to write a ledger transaction for an `askar-anoncreds` wallet, and then the anoncreds issuer listener would need to do that.

Reply:
Giving this some more thought, I think it's fair for us to define a general ledger API which includes functionality (like writing a ledger transaction) that isn't supported by all ledgers.  And then the implementation would just throw a "not implemented exception" as @jamshale has implemented here.

If we enforce an API including only those functions implemented by *all* ledgers then we're potentially limiting the functionality available to aca-py.  I.e. what if we get a new type of ledger later on that supported even *less* functionality?  Do we have to eliminate the API functions that this new ledger doesn't support?

Reply:
I took a closer look at this. The `txn_submit` is called from the endorser protocol and that's the only spot (that I identified quickly at least). My read on this is that this just shouldn't be an operation you access through the AnonCreds Interface. The limited interface isn't about holding back ACA-Py from additional features, it's about keeping the role of the AnonCreds registry clear and straightforward. If it's obvious that all AnonCreds implementations will require endorsement, then I think we can justify making changes to the interface to better support that. But for the moment, it's an Indy specific concern. I don't see a problem with calling up an instance of the Indy registry directly (instead of through the generic interface) and call additional functions on it that would otherwise be unavailable on the generic interface.

Reply:
Having the Endorser protocol call methods on the Indy registry directly I think is a more accurate representation of the dependency anyways since it is currently built explicitly for use by Indy networks.

Reply:
I think I understand. I was thinking you needed to access the indy_legacy registry through the interface. I'll try and implement it the way you descibed and then we can discuss again.